lib/pry/config/lazy.rb

```diff
@@ -25,6 +22,27 @@ def initialize(&block)
       def call
         @block.call
       end
+
+      #
+      # @example
+      #   c = Pry::Config.from_hash(foo: Pry.lazy { rand(100) }.memoize)
+      #   c.foo # => 21
+      #   c.foo # => 21
+      #   c.foo # => 21
+      #
+      # @return [Pry::Lazy]
+      #   A {Pry::Lazy}-like object with a notable difference: it memoizes its
+      #   return value after being called for the first time.
+      #
+      def memoize
+        Class.new(self.class) do
```

Comment:
This doesn't look like a simplification because creating classes at runtime *is* metaprogramming.

If memoization is really needed, the Hash class can handle that for free:

```rb
config = Hash.new { |h, key| h[key] = rand }
config[:foo]
#=> 0.9741489650825949
config[:foo]
#=> 0.9741489650825949
```

Reply:
it sounds like you are implying meta programming is by its nature, always complicated? 
i don't think that's __always__ true. anyway, it is less complex than the previous solution for memoization (Pry::Config::Memoization), so still a simplification.

what are you proposing with the example of a Hash?

Reply:
are you forgetting we are dealing with `Pry::Lazy` here?
The hash example seems to have no relevancy,  Pry::Config already works exactly like that, just not with `Pry::Config::Lazy` objects.

Reply:
which would be exactly true for `Hash` as well.

Reply:
even if we choose to not use memoization in the end, i believe `Pry.lazy { rand(10000) }.memoize` is useful, how else is a user suppose to indicate memoization if we just do it for every key through a default proc?

Reply:
decided to remove it completely, given what we discussed:
https://github.com/pry/pry/pull/1897/commits/2dda6f539a2b4e2df0d46635526c700c4a8a2c7a